packages/spidermonkey.rb

```diff
@@ -0,0 +1,62 @@
+require 'package'
+
+class Spidermonkey < Package
+  description 'SpiderMonkey is Mozilla\'s JavaScript engine written in C and C++.'
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  version '84.0'
+  compatibility 'all'
+  source_url 'https://archive.mozilla.org/pub/firefox/releases/84.0/source/firefox-84.0.source.tar.xz'
+  source_sha256 '23273ef0165b243f5d0908c38e7854d38070282c9b526e8d93b7503cd5f69138'
+
+  depends_on 'yasm'
+  depends_on 'rust' => :build
+  
+  def self.prebuild
+    puts
+    puts "Building spidermonkey requires at least 6.4 GB of space available on /mnt/stateful_partition.".lightblue
+    puts "It WILL fail without this space. Press Ctrl + C to cancel.".lightblue
```

Comment:
Why not determine free space available and abort if true?

Reply:
you're the rubyist! I didn't know how to do that in ruby, and i didn't wanna use df and sed/awk

Reply:
Something like this might show disk free: https://stackoverflow.com/a/37988177

Reply:
Just found that the [`android_studio`](https://github.com/skycocker/chromebrew/blob/master/packages/android_studio.rb) package is doing something like this